interactions/api/gateway/client.py

```diff
@@ -831,9 +831,11 @@ async def _send_packet(self, data: Dict[str, Any]) -> None:
             # This is because the ratelimiter limits already accounts for this.
             await self._ratelimiter.block()
 
-        self._last_send = perf_counter()
-        log.debug(packet)
-        await self._client.send_str(packet)
+        if self._client is not None:  # this mitigates against another edge case.
```

Comment:
Uhhh isn't it bad that we just don't send it? Maybe just make it while and wait until the client exist? Idk?

Reply:
The edge case here is that if the client doesn't exist whenever we reconnect, it would still send (during lib reconnection, client resets itself to None and there could be a very minor time frame where the heartbeat will still send in between heartbeat manager reset and client close. Also, iirc, the manager's supposed to reset on reconnect anyway. Of course, that could be optimised eventually so in terms of heartbeat, it still works/resets fine atm

(changing it to while wouldn't do anything)

(On current heartbeat testing, it still beats regularly, minus current gateway loop problems that seem to stem from didcord)